openlibrary/catalog/marc/parse.py

```diff
@@ -76,36 +82,28 @@ class SeeAlsoAsTitle(MarcException):
 )
 
 
-def read_dnb(rec):
+def read_dnb(rec: MarcBase) -> dict[str, list[str]] | None:
     fields = rec.get_fields('016')
     for f in fields:
-        (source,) = f.get_subfield_values('2') or [None]
-        (control_number,) = f.get_subfield_values('a') or [None]
+        (source,) = f.get_subfield_values('2') or ['']
+        (control_number,) = f.get_subfield_values('a') or ['']
         if source == DNB_AGENCY_CODE and control_number:
             return {'dnb': [control_number]}
+    return None
 
 
-def read_issn(rec):
+def read_issn(rec: MarcBase) -> dict[str, list[str]] | None:
     fields = rec.get_fields('022')
     if not fields:
-        return
-    found = []
-    for f in fields:
-        for k, v in f.get_subfields(['a']):
-            issn = v.strip()
-            if issn:
-                found.append(issn)
-    return {'issn': found}
+        return None
+    return {'issn': [v for f in fields for v in f.get_subfield_values('a')]}
```

Comment:
I understand that this is how the existing code is structured, so not asking for it to be changed, but it seems a little odd to return a single keyed dict. A more traditional approach would return an array for the caller to add to the dict with the key of its choice.

Reply:
Yes I agree, this struck me as odd when I first saw it too, then I saw why the code was doing that and went with it -- `identifiers` is one of the few fields in the OL model that takes a list of dicts, and these return values can be added directly to the list. 

Now that you bring it up again, it still seems unusual. Possibly it allows the values to be handled the same later one, but there still should be a way to refactor it to be more sensible. I'll take another look and see if there is an obvious clean up.

Reply:
The return values of `read_lccn()` and `read_dnb()` are structured this way to feed into `update_edition()` in this block:

https://github.com/internetarchive/openlibrary/blob/9e7e76a42bc071d137aad44a86859ae5bcc74d96/openlibrary/catalog/marc/parse.py#L672-L687

Still agree it's odd, but I'm less inclined to change it now. I'm not sure exactly what `update_edition()` is giving us, but it looks like it might require a bigger refactor / rethink.

Reply:
Not planning to change this code structure further. Possibly a refactoring task for later.